### ISSUE TITLE
conda build produces a corrupted output rendered meta.yaml

### ISSUE THREAD

**Issue by mwiebe** [conda/conda-build#1588]
What appears to be happening is the code https://github.com/conda/conda-build/blob/master/conda_build/build.py#L221 is overwriting perfectly fine build deps with a function object that gets rendered in some yaml escape syntax. The repr of the bad `build_deps` is `<bound method Environment.package_specs of <conda_build.environ.Environment object at 0x7fafdb8c6050>>`

If I fix up that line 221 to this (adding the function call):

```
    build_deps = environ.Environment(m.config.build_prefix).package_specs()
```

the corrupted yaml shown below becomes

```
   build:
       - toolchain 1.0 vfxp2016
       - cmake 3.7.1 vfxp2016_1
       - vfxp2016 2.0 4
       - zlib 1.2.8 vfxp2016_9
   run:
       - toolchain * vfxp2016
```

The output from `conda render` which is as expected is:

```
requirements:
    build:
        - vfxp2016
        - cmake
        - zlib
    run:
        - toolchain * vfxp2016
```

The output from `conda build` when it produces its final rendered output gets corrupted to:

```
requirements:
    build: !!python/object:__builtin__.instancemethod {}
    run:
        - toolchain * vfxp2016
```

A corrupted output from another platform is:

```
requirements:
    build: !!python/object/apply:builtins.getattr
        - !!python/object:conda_build.environ.Environment
            _meta: C:\Anaconda3\conda-bld\ptex_1482084865450\_b_env\conda-meta
            _packages: {}
            path: C:\Anaconda3\conda-bld\ptex_1482084865450\_b_env
        - package_specs
    run:
        - toolchain * msvc2015rel
```

Curiously the vfxp2016 one fails to parse, while the msvc2015rel one succeeds.

**Comment by msarahan** [comment]
Reopening to remind me to write a test for this.  Thanks @mwiebe

**Comment by sscherfke** [comment]
Current conda-build versions now produce an empty list for build requirements.
Input recipe:
```
requirements:
  build:
    - python
    - setuptools
```
built package -> info/recipe/meta.yaml:
```
requirements:
  build: []
```

**Comment by msarahan** [comment]
Thanks, I'll look into this ASAP.

**Comment by msarahan** [comment]
fixed and tested in #1585
